common/src/main/java/com/hibiscusmc/hmccosmetics/cosmetic/CosmeticHolder.java

```diff
@@ -0,0 +1,75 @@
+package com.hibiscusmc.hmccosmetics.cosmetic;
+
+import com.google.common.collect.ImmutableCollection;
+import com.hibiscusmc.hmccosmetics.user.CosmeticUser;
+import org.bukkit.Color;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+/**
+ * Represents any object that can wear/hold cosmetics,
+ * users, for example.
+ */
+public interface CosmeticHolder {
+    @Nullable Cosmetic getCosmetic(CosmeticSlot slot);
+
+    ImmutableCollection<Cosmetic> getCosmetics();
+
+    void addCosmetic(@NotNull Cosmetic cosmetic, @Nullable Color color);
+
+    default void addCosmetic(@NotNull Cosmetic cosmetic) {
+        addCosmetic(cosmetic, null);
+    }
+
+    default void removeCosmetics() {
+        // Small optimization could be made, but Concurrent modification prevents us from both getting and removing
+        for (CosmeticSlot slot : CosmeticSlot.values().values()) {
+            removeCosmeticSlot(slot);
+        }
+    }
+
+    void removeCosmeticSlot(CosmeticSlot slot);
+
+    default void removeCosmeticSlot(Cosmetic cosmetic) {

```

Comment:
Could you extend the annotations (such as @ NotNull for variables). With this class being in use a lot in third party plugins, want to clear what each method will/won't accept.